src/material/datepicker/datepicker-input.ts

```diff
@@ -255,6 +269,8 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
     this._localeSubscription = _dateAdapter.localeChanges.subscribe(() => {
       this.value = this.value;
     });
+
+    this._value = new MatSingleDateSelectionModel(this._dateAdapter, null);
```

Comment:
Move this up, immediately below the error checks so that we guarantee it's set before doing any real work